src/test/java/net/snowflake/client/jdbc/ArrowBatchesTest.java

```diff
@@ -0,0 +1,241 @@
+package net.snowflake.client.jdbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import net.snowflake.client.core.SFArrowResultSet;
+import org.apache.arrow.vector.BigIntVector;
+import org.apache.arrow.vector.DecimalVector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.SmallIntVector;
+import org.apache.arrow.vector.TinyIntVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ArrowBatchesTest extends BaseJDBCWithSharedConnectionIT {
+  @Before
+  public void setUp() throws Exception {
+    try (Statement statement = connection.createStatement()) {
+      statement.execute("alter session set jdbc_query_result_format = 'arrow'");
+    }
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    try (Statement statement = connection.createStatement()) {
+      statement.execute("alter session unset jdbc_query_result_format");
+    }
+  }
+
+  private static void assertNoMemoryLeaks(ResultSet rs) throws SQLException {
+    assertEquals(
+        ((SFArrowResultSet) rs.unwrap(SnowflakeResultSetV1.class).sfBaseResultSet)
+            .getAllocatedMemory(),
+        0);
+  }
+
+  @Test
+  public void testMultipleBatches() throws Exception {
+    Statement statement = connection.createStatement();
+    ResultSet rs =
```

Comment:
Please move statement and result set to try-with-resources

Reply:
Will do